mobile/src/home-view.js

```diff
@@ -109,7 +109,12 @@ class HomeView extends PureComponent {
     this.props.fbc.database.private
       .adminRef('adminUrl')
       .on('value', data => this.setState({ adminUrl: data.val() || undefined }))
-    this.adminSessionsRef().on('value', data => this.setState({ adminSessions: data.val() || {} }))
+    this.adminSessionsRef().on('value', data => {
+      this.setState({ adminSessions: data.val() || {} })
+      if (this.state.adminSessionId && !data.val()[this.state.adminSessionId]){
```

Comment:
When setting state based on previous state, use the `setState(updaterFunction)` version: https://reactjs.org/docs/react-component.html#setstate

Otherwise, this is not guaranteed to be correct and may break as React makes performance improvements.

Reply:
I'm not sure I understand whats necessary as one variable is locally based and one is set by firebase. This is essentially for if you are viewing the trivia controls then its deleted that you would be removed from the game control view.

Reply:
`this.state` is not guaranteed to give you the most up-to-date value (because React may batch the `setState`s.  However, calling like `this.setState(prevState => { /* return some state updates */ })` will give you a `prevState` that has taken into account all previous `setState`s